### ISSUE TITLE
0.29.0 never becomes ready

### ISSUE THREAD

**Issue by dgn** [operator-framework/operator-lifecycle-manager#3419]
## Bug Report

**What did you do?**
We regularly install the olm-operator on KinD using `operator-sdk olm install`. This has started failing since 0.29.0 has come out. 0.28.0 installs just fine. We're using the latest version of `operator-sdk` (v1.37.0)

Here's a log:
```
INFO[0010] Creating OLM resources...                    
INFO[0010]   Creating Namespace "olm"                   
INFO[0012]   Namespace "olm" created                    
INFO[0012]   Creating Namespace "operators"             
INFO[0013]   Namespace "operators" created              
[...snip...]
INFO[0024]   Creating CatalogSource "olm/operatorhubio-catalog" 
INFO[0025]   CatalogSource "olm/operatorhubio-catalog" created 
INFO[0025] Waiting for deployment/olm-operator rollout to complete 
INFO[0026]   Waiting for Deployment "olm/olm-operator" to rollout: 0 of 1 updated replicas are available 
FATA[0120] Failed to install OLM version "latest": deployment/olm-operator failed to rollout: context deadline exceeded 
```

The containers fail to start with `CreateContainerConfigError`. Here's the relevant event log:

```
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Warning  Failed     5s (x11 over 2m1s)  kubelet            Error: container has runAsNonRoot and image will run as root (pod: "olm-operator-7b6ffc567-nnlm7_olm(5b55cb5e-8d14-4b71-abd7-8fd8644c28be)", container: olm-operator)
  Normal   Pulled     5s (x10 over 2m)    kubelet            Container image "quay.io/operator-framework/olm@sha256:cfdda06700fae2962ac89e677c1968517f8c5fae838ca31774b909f545d7be37" already present on machine
```

**What did you expect to see?**
A working installation of olm-operator using `operator-sdk olm install`

**What did you see instead? Under which circumstances?**
Deployment never becomes ready.

**Environment**
* operator-lifecycle-manager version:

0.29.0

* Kubernetes version information:

v1.29.2

* Kubernetes cluster kind: 

KinD cluster

**Possible Solution**
Looks like `container has runAsNonRoot and image will run as root` from the event log is the issue, but there might be more.

**Additional context**

**Comment by jmazzitelli** [comment]
FWIW I see the same problem on minikube:

```
$ minikube version
minikube version: v1.34.0
commit: 210b148df93a80eb872ecbeb7e35281b3c582c61
```

**Comment by jmazzitelli** [comment]
> We regularly install the olm-operator on KinD using `operator-sdk olm install`

I also want to say that we (Kiali team) start it a different way, but get the same error. Here's how we deploy OLM: https://github.com/kiali/kiali/blob/v2.0.0/make/Makefile.olm.mk#L249

### REPLICATION PROCEDURES TO SEE THE ERROR:

Basically, download the install.sh and pipe it to bash.

You can replicate the problem easily on minikube:

1. Start minikube
2. Download OLM v0.29.0 install script and pipe to bash:
```
curl -sL https://github.com/operator-framework/operator-lifecycle-manager/releases/download/v0.29.0/install.sh | bash -s v0.29.0
```
3. Notice script hangs and eventually times out waiting:
```
...
Waiting for deployment "olm-operator" rollout to finish: 0 of 1 updated replicas are available...
```
4.  Look at the OLM pods fail to start
```
$ kubectl get pods -n olm
NAME                                READY   STATUS                       RESTARTS   AGE
catalog-operator-568c6d5989-75psv   0/1     CreateContainerConfigError   0          37s
olm-operator-7c4756f796-mdn8p       0/1     CreateContainerConfigError   0          37s
```

```
$ kubectl describe pods -n olm -l app=olm-operator
  Normal   Scheduled  99s                default-scheduler  Successfully assigned olm/olm-operator-7c4756f796-mdn8p to minikube
  Normal   Pulling    98s                kubelet            Pulling image "quay.io/operator-framework/olm@sha256:cfdda06700fae2962ac89e677c1968517f8c5fae838ca31774b909f545d7be37"
  Normal   Pulled     78s                kubelet            Successfully pulled image "quay.io/operator-framework/olm@sha256:cfdda06700fae2962ac89e677c1968517f8c5fae838ca31774b909f545d7be37" in 19.958s (19.958s including waiting). Image size: 235113812 bytes.
  Warning  Failed     10s (x7 over 78s)  kubelet            Error: container has runAsNonRoot and image will run as root (pod: "olm-operator-7c4756f796-mdn8p_olm(fd01319b-d680-48e8-8d03-5a66385ae1d9)", container: olm-operator)
  Normal   Pulled     10s (x6 over 78s)  kubelet            Container image "quay.io/operator-framework/olm@sha256:cfdda06700fae2962ac89e677c1968517f8c5fae838ca31774b909f545d7be37" already present on machine
```

**Comment by tyxeron** [comment]
We face the same on k3d:
```
$ k3d version                   
k3d version v5.7.4
k3s version v1.30.4-k3s1 (default)
```
We used the same procedure to install OLM as @jmazzitelli and faced exactly the same issue.

**Comment by acornett21** [comment]
The issue appears to be that the `sha` in the `olm.yaml` file points to to a sha for `v0.19.0` instead of `v0.29.0`

```
containerStatuses:
  - image: quay.io/operator-framework/olm@sha256:cfdda06700fae2962ac89e677c1968517f8c5fae838ca31774b909f545d7be37
    imageID: ""
    lastState: {}
    name: olm-operator
    ready: false
    restartCount: 0
    started: false
    state:
      waiting:
        message: 'container has runAsNonRoot and image will run as root (pod: "olm-operator-7b6ffc567-4svgn_olm(96ffa7d3-4fcb-4ea0-bd19-bae6baf40ea4)",
          container: olm-operator)'
        reason: CreateContainerConfigError
```

```
crane digest quay.io/operator-framework/olm:v0.19.0
sha256:cfdda06700fae2962ac89e677c1968517f8c5fae838ca31774b909f545d7be37
```

**Comment by m1kola** [comment]
[OLM_VERSION](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/OLM_VERSION) in the repo root still contains `0.19.0`. It seems like `make release` depends on this:

https://github.com/operator-framework/operator-lifecycle-manager/blob/06edf283934b43a1d8341e24181606f072cf6a9d/Makefile#L333

But it is not clear to me how all the releases after `0.19.0` were created correctly.

**Comment by m1kola** [comment]
Ok, I think I understand the issue now. In the release workflow we pass the actual version here:

https://github.com/operator-framework/operator-lifecycle-manager/blob/06edf283934b43a1d8341e24181606f072cf6a9d/.github/workflows/goreleaser.yaml#L63

But we no longer read it in the `Makefile` after [this change](https://github.com/operator-framework/operator-lifecycle-manager/pull/3280/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L234-L241) in #3280

**Comment by acornett21** [comment]
@m1kola Thanks for diving into this, It seems like #3280 should have been broken into more manageable pieces so it could have been better reviewed. A PR with ~1800 changes can't be reviewed properly IMO.

Is the plan to merge in #3426 and re-release `0.29.0`? ie delete the current release, images, etc, and re-do a new `0.29.0` release?

**Comment by m1kola** [comment]
> It seems like #3280 should have been broken into more manageable pieces so it could have been better reviewed. A PR with ~1800 changes can't be reviewed properly IMO.

I'm a big fan of small PRs and this one could've probably been spliced into smaller ones. But to be fair - most of the file changes are in vendor directory.


> Is the plan to merge in #3426 and re-release `0.29.0`? ie delete the current release, images, etc, and re-do a new `0.29.0` release?

I'm not a big fan of replacing releases and I'm not sure that our tooling supports this. Probably will be better to create a new release and potentially remove the old one. Once #3426 is ready to go in I'll think about it.

**Comment by acornett21** [comment]
> I'm not a big fan of replacing releases and I'm not sure that our tooling supports this. Probably will be better to create a new release and potentially remove the old one. Once #3426 is ready to go in I'll think about it.

Maybe all that's needed is to remove the assets from `0.29.0` and make a note that it was broken, so the assets were removed. Not sure if that's a more confusing experience or not.

**Comment by jon-nfc** [comment]
is it only olm.yaml that's affected or are crds.yaml affected too?

**Comment by jmazzitelli** [comment]
So -- what's the final solution? Is 0.29.0 fixed or is another release coming?

**Comment by m1kola** [comment]
We had a discussion in K8s slack in [this thread](https://kubernetes.slack.com/archives/C0181L6JYQ2/p1730299622108809) and it was decided that the best course of action is:

1. Add a note to [0.29.0 release](https://github.com/operator-framework/operator-lifecycle-manager/releases/tag/v0.29.0) and remove artefacts from it. This part is already done.
2. Release a new version. I'll be looking into this today.

**Comment by m1kola** [comment]
We have a number of issues with release automation tooling which we need to resolve. I fixed some, but we are still not 100% recovered.

So far I merged the following:
* https://github.com/operator-framework/operator-lifecycle-manager/pull/3426
* https://github.com/operator-framework/operator-lifecycle-manager/pull/3428
* https://github.com/operator-framework/operator-lifecycle-manager/pull/3429

But it looks like our manifests get populated with arch specific image refs based on the CI arch instead of a ref of `v0.30.0`. Still working on it.

I also noticed that this issue got automatically closed. Re-opening for now.

**Comment by m1kola** [comment]
We just released [v0.30.0](https://github.com/operator-framework/operator-lifecycle-manager/releases/tag/v0.30.0).

I'm sorry it took so long to address this: we had several issues due to changes in the release automation which needed to be addressed and while fixing these we also discovered [some other issues](https://github.com/operator-framework/operator-lifecycle-manager/pull/3434).
